sync/sync_head.go

```diff
@@ -189,7 +189,10 @@ func isExpired[H header.Header[H]](header H, period time.Duration) bool {
 	return !expirationTime.After(time.Now())
 }
 
-// isRecent checks if header is recent against the given blockTime.
-func isRecent[H header.Header[H]](header H, blockTime time.Duration) bool {
-	return time.Since(header.Time()) <= blockTime+blockTime/2 // add half block time drift
+// isRecent checks if header is recent against the given recency threshold.
+func isRecent[H header.Header[H]](header H, blockTime, recencyThreshold time.Duration) bool {
+	if recencyThreshold == 0 {
+		recencyThreshold = blockTime + blockTime/2 // allow some drift by adding additional buffer of 1/2 of block time
```

Comment:
nit: It would be cleaner to make it a method on Syncer or Params, as it now has access to fields from it.